packages/@angular/cli/tasks/serve.ts

```diff
@@ -108,7 +108,7 @@ export default Task.extend({
       stats: statsConfig,
       inline: true,
       proxy: proxyConfig,
-      compress: serveTaskOptions.target === 'production',
+      compress: serveTaskOptions.target === 'production' && serveTaskOptions.compress,
```

Comment:
We need to make sure, that the compress option will only be available in combination with -t production, but this behavior can be changed in general

@filipesilva